lib/compiler/src/beam_doc.erl

```diff
@@ -313,6 +311,21 @@ preprocessing(AST, State) ->
                        end,
    foldl(PreprocessingFuns, State, AST).
 
+extract_deprecated({attribute, Anno, deprecated, Deprecations}, State)
+  when is_list(Deprecations) ->
+    lists:foldl(fun(D, S) ->
+                        extract_deprecated({attribute, Anno, deprecated, D}, S)
+                end, State, Deprecations);
+extract_deprecated({attribute, Anno, deprecated_type, Deprecations}, State)
+  when is_list(Deprecations) ->
+    lists:foldl(fun(D, S) ->
+                        extract_deprecated({attribute, Anno, deprecated_type, D}, S)
+                end, State, Deprecations);
+extract_deprecated({attribute, Anno, deprecated_callback, Deprecations}, State)
+  when is_list(Deprecations) ->
+    lists:foldl(fun(D, S) ->
+                        extract_deprecated({attribute, Anno, deprecated_callback, D}, S)
+                end, State, Deprecations);
```

Comment:
Since this code is exactly the same in these three places, would it make sense to re-write as:
```erlang
extract_deprecated({attribute, Anno, DeprecatedAtom, Deprecations}, State)
  when is_list(Deprecations), 
           (DeprecatedAtom =:= deprecated; 
            DeprecatedAtom =:= deprecated_type; 
            DeprecatedAtom =:= deprecated_callback) ->
    lists:foldl(fun(D, S) ->
                        extract_deprecated({attribute, Anno, deprecated, D}, S)
                end, State, Deprecations);
```

It is a tiny bit overloaded in the guard but we gain visibility that the cases are handled in the same way